report.md

```diff
@@ -2,6 +2,24 @@
 
 ## Subsystem 1 (MFA)
 
+Multifactor authentication will apply to all users and will aim for a balance between confidentiality and availability.
+
+###First Factor
+The first step of authentication will involve a username and password. The username will be stored as a hash. The password will be stored as a hash with salt. The salt for each user will be unique and will be generated upon registration. The salt will be stored as plain text.
+
+###Second Factor
+The second factor will be triggered on the completion of the first factor. On passing the first stage a random string will be generated by the server and stored temporarily in the database with the associated user in hash form. An sms will be sent to the user based on the registered phone number with the digits. The user will be redirected to another form to key in the digits. He will have 2 minutes to do so after which the login process will fail and the temporary string will be replaced with nonces. On success the user will receive a session and be able to login and the string will be replaced with nonces.
+
+###Third factor for special actions
+The third factor will be applicable only to administrators. The administrators will have a specially assigned BLE token with a private symmetric??? key. When logging in as an administrator, the token will have to be present and scanned before gaining access to the account on top of the first and second factor.
+
+###Services and tools to be used
+GraphQL and PHP for the initial registration. PHP crypto libraries will be used to encrypt the usernames,passwords and other data.

```

Comment:
Previously I and @gjlam95 were discussing that there's no need for PHP cos PHP is meant to be used to connect to the database, but that will be handled by Spring instead.

Following your comment https://github.com/IFS4205-2018-Sem1-Team1/design-report/pull/7#issuecomment-418690092, there's 2 ways to support file upload:
1. Apollo (using js)
1. PHP

I googled a bit and found [this](https://stackoverflow.com/questions/48050509/graphql-spring-boot-upload-binary), basically we can use Spring Boot to upload binaries also.

I'm thinking if possible, we should try to have all the data transfer on the same "side" instead of splitting over Client & Server.

What do you think? :)

Reply:
Ok got it, I will check whether there's a way to transfer binaries confidentially. The crypto library basically ensures confidentiality right?